Automattic/woocommerce-payments
# Customer Multi-Currency Onboarding

Fixes #1778

#### Changes proposed in this Pull Request


Add customer multi-currency onboarding screen

This PR adds a onboarding wizard for the customer multi-currency feature, which enables the user to select currencies from a searchable checkbox list, with a predefined "Recommended currencies" section, displays the already enabled currencies (store's default currency and the currencies possibly set by the UPE methods) on top, set store settings such as automatic currency switching by geolocation and displaying the currency switcher on breadcrumb section when the user uses the Storefront theme, and displays a summary section after the wizard completes. It also includes a preview modal, which the user can see what happens when they change the store settings before saving them.  

This section is currently accessible via the task list and the note on the WooCommerce overview (home) screen. Once the onboarding is completed, the links will redirect the user to the Multi-Currency Settings tab in the WooCommerce > Settings page.


The screens look like this:

##### WooCommerce home page <strike>task list item</strike> and the note

![image](https://user-images.githubusercontent.com/3295/132408699-76e445ea-f87c-4bdc-99b2-deb28d3a37f1.png)

**Note**: Task list item is hidden for now. It can be set back to visible by changing the `visible` value to `true`: https://github.com/Automattic/woocommerce-payments/blob/ec8a91d7a03bad9b0b503ce3dd7afe6bc9767373/client/multi-currency-setup/task.js#L33-L35

##### Add Currencies Task

![image](https://user-images.githubusercontent.com/3295/132408735-2222aa3b-4b5b-4e38-ad7a-6decf9aad844.png)

##### Store Settings Task

![image](https://user-images.githubusercontent.com/3295/132408793-4998cb98-7e7e-4ed2-bccb-1bacefde8932.png)

##### Preview Modal

![image](https://user-images.githubusercontent.com/3295/132408896-fbdcaa5a-31ff-4225-bea3-6a8247cd4064.png)

##### Setup Complete Screen 

![image](https://user-images.githubusercontent.com/3295/132408945-4c74bf37-7374-41cf-913c-e702b513285e.png)

#### Testing instructions


##### Automated Testing:
- Run `npm run test:js multi-currency-setup`
##### Manual Testing:
- Open *WooCommerce > Home* page and click **Sell worldwide in multiple currencies** link in the **Things to do next** task list, or click the **Set up now** button in the **Sell worldwide in multiple currencies** titled note.
- Check the currency in the description matches the store currency.
- Follow the wizard tasks
- Check if the button shows the right amount 
- Check if WooCommerce Multi-Currency tab settings match with the settings you've chosen in the onboarding wizard.
- Check if you can re-enter the onboarding wizard once you've completed the tasks.
##### Upe enabled manual testing
- Enable some UPE methods to pre-add currencies to the enabled currencies list
- Open the onboarding wizard
- Check if those pre-added currencies available for selection, they shouldn't be on the checkbox list, they should be displayed as text above the currency selection form.

#### Status Reset
Once you completed the onboarding wizard, you can delete the `wcpay_multi_currency_setup_completed` option to reset the onboarding status. If you want the whole multi-currency states reverted back, you can run this query:

```sql
DELETE FROM wp_options where option_name like '%multi_currency%'
```



-------------------


**Post merge**
